firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -1844,63 +1844,120 @@ class BrowserViewController: UIViewController,
+    func addressAutofillNimbusFeatureFlag() -> Bool {
+        return featureFlags.isFeatureEnabled(.addressAutofill, checking: .buildOnly)
+    }
+
+    /// Returns whether address autofill settings are enabled based on user defaults.
+    func addressAutofillSettingsUserDefaultsIsEnabled() -> Bool {
+        let userDefaults = UserDefaults.standard
+        let keyAddressAutofill = PrefsKeys.KeyAutofillAddressStatus
+
+        return (userDefaults.object(forKey: keyAddressAutofill) as? Bool ?? true)
+    }
+
+    /// Sets up credit card autofill handling.
+    private func creditCardAutofillSetup(_ tab: Tab, didCreateWebView webView: WKWebView) {
         let formAutofillHelper = FormAutofillHelper(tab: tab)
         tab.addContentScript(formAutofillHelper, name: FormAutofillHelper.name())
+
+        // Closure to handle found field values for credit card and address fields
         formAutofillHelper.foundFieldValues = { [weak self] fieldValues, type, frame in
-            guard let tabWebView = tab.webView,
-                  let type = type,
-                  userDefaults.object(forKey: keyCreditCardAutofill) as? Bool ?? true
-            else { return }
+            guard let strongSelf = self, let tabWebView = tab.webView else { return }
+
+            // Handle different field types
+            switch fieldValues.fieldValue {
+            case .address:
+                guard let addressPayload = fieldValues.fieldData as? UnencryptedAddressFields,
+                      strongSelf.addressAutofillSettingsUserDefaultsIsEnabled(),
+                      strongSelf.addressAutofillNimbusFeatureFlag(),
+                      let type = type else { return }
+
+                // Handle address form filling or capturing
+                switch type {
+                case .fillAddressForm:
+                    strongSelf.displayAddressAutofillAccessoryView(tabWebView: tabWebView)
+                case .captureAddressForm:
+                    break // No action needed for capturing address form
+                default:
+                    break
+                }
 
-            // FXIOS-7150: Telemetry for form input regardless of feature flag status
-            if type == .formInput {
-                TelemetryWrapper.recordEvent(category: .action,
-                                             method: .tap,
-                                             object: .creditCardFormDetected)
+            case .creditCard:
+                guard let creditCardPayload = fieldValues.fieldData as? UnencryptedCreditCardFields,
+                      let type = type,
+                      strongSelf.autofillCreditCardSettingsUserDefaultIsEnabled() else { return }
+
+                // Record telemetry for credit card form detection
+                if type == .formInput {
+                    TelemetryWrapper.recordEvent(category: .action, method: .tap, object: .creditCardFormDetected)
```

Comment:
nit: indent, we usually go with the following formatting for multiple params
```suggestion
                    TelemetryWrapper.recordEvent(category: .action,
                                                 method: .tap,
                                                 object: .creditCardFormDetected)
```